### PR TITLE
Add function to take an input type and convert it to a union.

### PR DESCRIPTION
**Pull request by robshakir**
This PR adds a change which will take an input type and convert it to the struct which implements an interface for a union field. 

In addition, it updates the OpenConfig example modules to the latest version and adds a script to rebuild the `exampleoc` package.

### CODE REVIEW

File: ygen/gogen.go [hunk 1]
```diff
@@ -515,7 +516,23 @@ type {{ $intfName }}_{{ $typeName }} struct {
...
+func (t *{{ .ParentReceiver }}) To_{{ .Name }}(i interface{}) ({{ .Name }}, error) {
+	switch v := i.(type) {
+	{{ range $typeName, $type := .Types -}}
+	case {{ $type }}:
+		return &{{ $intfName }}_{{ $typeName }}{v}, nil
+	{{ end -}}
+	default:
+		return nil, fmt.Errorf("cannot convert %v to {{ .Name }}, unknown union type, got: %T, want: 
+		{{- range $typeName, $type := .Types -}}{{ $type }} {{ end -}}
+		", i, i)
```
Comment (ostromart) [inline ygen/gogen.go]: want any of: [type1, type2...] would make the error clearer.

File: ygen/testdata/schema/openconfig-options-compress-fakeroot.formatted-txt [hunk 2]
```diff
@@ -128,6 +128,20 @@ type Bgp_Neighbor_EnabledAddressFamily_Union_Uint32 struct {
...
+// cannot be converted to a type within the union.
+func (t *Bgp_Neighbor) To_Bgp_Neighbor_EnabledAddressFamily_Union(i interface{}) (Bgp_Neighbor_EnabledAddressFamily_Union, error) {
+	switch v := i.(type) {
+	case E_OpenconfigOptions_AFI:
+		return &Bgp_Neighbor_EnabledAddressFamily_Union_E_OpenconfigOptions_AFI{v}, nil
+	case uint32:
+		return &Bgp_Neighbor_EnabledAddressFamily_Union_Uint32{v}, nil
+	default:
+		return nil, fmt.Errorf("cannot convert %!v(MISSING) to Bgp_Neighbor_EnabledAddressFamily_Union, unknown union type, got: %!T(MISSING), want:E_OpenconfigOptions_AFI uint32 ", i, i)
+	}
```
Comment (ostromart) [inline ygen/testdata/schema/openconfig-options-compress-fakeroot.formatted-txt]: Generation problem @MISSING, here and below?

Comment (robshakir) [reply ygen/testdata/schema/openconfig-options-compress-fakeroot.formatted-txt]: Sorry - issue with copy and pasting expected output as I vetted these. Will fix.

File: ygen/gogen.go (hunk 1 above)
Comment (robshakir) [reply ygen/gogen.go]: Done - required a minor tweak to the input struct based on limitations in Go templating.

**Pull request merged**